model/src/main/java/io/quarkus/qe/model/QuarkusExtension.java

```diff
@@ -11,23 +22,31 @@ public void setName(String name) {
         this.name = name;
     }
 
+    public String getVersion() {
+        return version;
+    }
+
+    public void setVersion(String version) {
+        this.version = version;
+    }
+
     @Override
     public boolean equals(Object obj) {
         if (obj == this)
             return true;
         if (!(obj instanceof QuarkusExtension))
             return false;
         QuarkusExtension other = (QuarkusExtension) obj;
-        return name.equals(other.name);
+        return (version == null) ? name.equals(other.name) : name.equals(other.name) && version.equals(other.version);
     }
 
     @Override
     public final int hashCode() {
-        return name.hashCode();
+        return (name + "_" + version).hashCode();
     }
 
     @Override
     public String toString() {
-        return name;
+        return name + "_" + version;
```

Comment:
This will look ugly when the version is null.
Please, use ":" character for the toString method.